### PR TITLE
Fix deploy

### PR DESCRIPTION
**Pull request by brettstack**
This fixes errors when running `npm run deploy:dev` and gets CI working again (at least it's working for PR; we'll see if more work is needed when we merge to master).

### CODE REVIEW

File: templates/resources.yml
Comment (brettstack) [inline templates/resources.yml:100]: @bboure Could you please take a look at this and replacing the Cognito Authorizer stuff below to work with AppSync?

File: templates/graphql.yml
Comment (brettstack) [inline templates/graphql.yml:56]: @bboure is this the best way to do this?

File: templates/dynamodb.yml
Comment (brettstack) [inline templates/dynamodb.yml:6]: @bboure anything else you can think of we'll need to do to get non-hardcoded names working?

File: packages/ui/package.json
Comment (brettstack) [inline packages/ui/package.json:8]: The serverless-amplify-plugin expects `build` command to do everything that's necessary for build which is why we can't call this "build-export" and leave "build" purely as "next build"

File: packages/api/functions/cognito/post-authentication.js
Comment (brettstack) [inline packages/api/functions/cognito/post-authentication.js:16]: This is a no-op currently. Need to update it to insert user if they don't currently exist in Dynamo

File: packages/api/functions/cognito/auto-confirm-user.js
Comment (brettstack) [inline packages/api/functions/cognito/auto-confirm-user.js:7]: @sanathkr for product input. I don't care much for verify email. It adds friction to real users. If someone really wants to create fake accounts, they're going to create fake accounts.

File: .github/workflows/on-push-to-master.yaml
Comment (brettstack) [inline .github/workflows/on-push-to-master.yaml:26]: Ideally, we'd leave this as "high", but in reality, it causes way too many blockers when waiting on dependencies to update their downstream dependencies.

File: .github/workflows/on-pr.yaml
Comment (brettstack) [inline .github/workflows/on-pr.yaml:49]: Using event.number (the PR ID) instead of the branch name because long branch names cause Serverless Framework/CloudFormation long logical resource id errors.

**Comment by pipesorter** [comment]
The site is live, yay! 🎉 🎉

**Review by pipesorter — DISMISSED** [review]

**Comment by brettstack** [comment]
> The site is live, yay! 🎉 🎉

Not yet. This is just a preview/sandbox stack that will get auto-deleted once this PR is merged😉 Hopefully all goes well with GH Actions on merge, though these things rarely work first time 😅

File: templates/dynamodb.yml
Comment (bboure) [reply templates/dynamodb.yml:6]: @brettstack no, it should keep working like [this](https://github.com/RareCamp/RareCamp/blob/master/templates/graphql.yml#L49-L51)

File: templates/graphql.yml
Comment (bboure) [reply templates/graphql.yml:56]: @bboure no, the best way is as it was. Use `functionName: getDisease` where the value is the logical name (key) of the function, [here](https://github.com/RareCamp/RareCamp/blob/master/templates/functions.yml#L1)

**Comment by pipesorter** [comment]
hahah, cool 😄

File: templates/resources.yml
Comment (bboure) [reply templates/resources.yml:100]: @brettstack I'll have a look asap.

**Comment by brettstack** [comment]
@pipesorter can I get another approval? I've now disabled the "Dismiss stale pull request approvals when new commits are pushed" setting

**Review by pipesorter — APPROVED** [review]

**Pull request merged**
